core/integration/module_test.go

```diff
@@ -2888,6 +2888,56 @@ class Foo {
 	}
 }
 
+func TestModuleScalarType(t *testing.T) {
+	// Verify use of a core scalar as an argument type.
+
+	t.Parallel()
+
+	type testCase struct {
+		sdk    string
+		source string
+	}
+	for _, tc := range []testCase{
+		{
+			sdk: "go",
+			source: `package main
+
+type Test struct{}
+
+func (m *Test) SayHello(platform Platform) string {
+	return "hello " + string(platform)
+}`,
+		},
+		{
+			sdk: "python",
+			source: `import dagger
+from dagger import function, object_type
+
+@object_type
+class Test:
+    @function
+    def say_hello(self, platform: dagger.Platform) -> str:
+        return f"hello {platform}"
```

Comment:
FTR, I've tested the Python change manually and it fixes the issue for both scalars and enums. Should add tests for enum as well.

But... I'm not sure this is the right test here. In Python, type checking doesn't do anything at runtime so `platform` here could be a string and everything runs without error. In fact, if I remove the error in Python when a scalar is used (reverting back to a normal string), this function passes without error.

However the generated client does have runtime type checks and the issue in https://github.com/dagger/dagger/issues/6466 is specifically about how codegen generates this function for the client. Maybe good to have a test create a dependent module to use this function?

Or just check if it creates the expected type in the API schema?

Reply:
In Go, this also passes on `main`, so it's only failing on the invalid platform error.

Reply:
That's intended - the first check is just a sanity check, the second one is the validation thing as described above.

Reply:
But what purpose does the sanity check have if it also passes without the changes in this PR?

Reply:
Because the PR has changed logic related to the parsing of scalars, and there were no tests prior to this that explicitly tested passing a valid scalar into a function that requires a scalar.

If this test isn't present, we are only testing that scalar validation can fail, not that scalar validation can succeed.

Reply:
Yeah, now that I see where the error comes from it does seem like a good enough test. 👍